hwles9-databases.sql

```diff
@@ -0,0 +1,133 @@
+--/* create user table and insert one user into that table
+Drop table if exists gebruiker
+CREATE TABLE gebruiker (
+	id SERIAL PRIMARY KEY,
+	username VARCHAR(255) NOT NULL,
+	password VARCHAR(255) CHECK(length(password) > 8),
+	address VARCHAR(255),
+	job VARCHAR(255),
+	wages INT,
+	holiday_days INT DEFAULT 25
+);
+
+INSERT INTO gebruiker (username, password, address, job, wages, holiday_days)
+VALUES ('benmij@icloud.com','hello1234','spuistraat 123','storemanager',3500,28);
+
+select * from gebruiker
+--*/
+
+
+--/*
+--Create tabel voor alle producten
+CREATE TABLE products (
+	id SERIAL PRIMARY KEY,
+	name VARCHAR(255),
+	brand VARCHAR(255) NOT NULL,
+	type VARCHAR(255),
+	price INT,
+	current_stock INT,
+	sold INT,
+	date_sold DATE
+);
+
+--Create televisie tabel
+CREATE TABLE television (
+	id SERIAL PRIMARY KEY,
+	height INT,
+	width INT,
+	screen_quality VARCHAR(255),
+	screen_type VARCHAR(255),
+	wifi BOOLEAN DEFAULT false,
+	smartTV BOOLEAN DEFAULT false,
+	voice_control BOOLEAN DEFAULT false,
+	HDR BOOLEAN DEFAULT false,
+	productId INT,
+	FOREIGN KEY (productId) REFERENCES products(id)
+);
+
+--Create tabel afstandsbediening
+Drop table if exists remoteController
+CREATE TABLE remoteController (
+	id SERIAL PRIMARY KEY,
+	smart BOOLEAN,
+	battery_type VARCHAR(255),
+	productId INT,
+	televisionId INT,
+	FOREIGN KEY (productId) REFERENCES products(id),
+	FOREIGN KEY (televisionId) REFERENCES television(id)
+);
+
+--Create tabel CiModule
+Drop table if exists ciModule
+CREATE TABLE ciModule (
+	id SERIAL PRIMARY KEY,
+	provider VARCHAR(255),
+	encodingCiModule VARCHAR(255),
+	productId INT,
+	televisionId INT,
```

Comment:
Een CI-module past op meerdere tv's, hetzelfde geldt voor wallbrackets. Simpelweg een television id werkt dan dus niet. Je zou in television een field `ciModuleId` kunnen maken aangezien elke tv maar één ci module kan hebben, of je kunt -zoals je voor wallbrackets sowieso zult moeten doen- een aparte tabel maken die tv's linkt aan andere apparaten. Dus bijv. een tabel `television_ciModule` met de fields `televisionId` en `ciModuleId`.